### DISCUSSION TITLE
django-pgschemas & celery

### DISCUSSION THREAD

**Discussion by ricou84** [lorinkoz/django-pgschemas#95] [Q&A] [answered]
Hello.

I have a working django application that is not multi tenant. This application use actively celery for time-consuming background task.
This celery task make many access to the database and update things like status and step.

Il want to integrate multi-tenancy mechanism with isolation for customer specific data set and some shared data set. 

Is there a way to make celery aware of multi tenancy in order to update table ? Celery worker will be shared between customer.

**Comment by lorinkoz** [comment, marked as answer]
The basic recommendation would be to pass the tenant as a parameter of the celery task, then perform the task in the context of that tenant.

**Reply by ricou84** [reply]
Hello thanks for your fast answer.
You think something like this https://www.google.com/amp/s/shulhi.com/2016/03/18/django-multi-tenant-postgres-schema-issue/amp/

So Django orm will not be usable anymore no?

**Reply by lorinkoz** [reply]
You can use the ORM again inside the task, you'd only need to get the tenant model instance from the tenant paramenter.

**Reply by RohitHazare** [reply]
Hi.. How to perform task in the context of tenant and use tenant model instance from tenant parameter.

**Reply by lorinkoz** [reply]
```
def task_do_something(tenant_id, *args, **kwargs):
    tenant = TenantModel.objects.get(pk=tenant_id)
    with tenant:
        ...  # Do the real something
```

**Reply by RohitHazare** [reply]
Thank you!! That helped.
